src/gui/torrentcreatordialog.cpp

```diff
@@ -300,16 +304,34 @@ void TorrentCreatorDialog::updateProgressBar(int progress)
 
 void TorrentCreatorDialog::updatePiecesCount()
 {
+    m_ui->buttonCalcTotalPieces->setEnabled(false);
+    m_ui->labelTotalPieces->setText(tr("Calculating..."));
+
     const Path path = m_ui->textInputPath->selectedPath();
 #ifdef QBT_USES_LIBTORRENT2
-    const int count = BitTorrent::TorrentCreator::calculateTotalPieces(
-        path, getPieceSize(), getTorrentFormat());
+    auto *calculator = QRunnable::create([path, pieceSize = getPieceSize(), format = getTorrentFormat(), receiver = QPointer<TorrentCreatorDialog>(this)]() {
+        const int count = BitTorrent::TorrentCreator::calculateTotalPieces(path, pieceSize, format);
+        if (receiver) {
+            QMetaObject::invokeMethod(receiver.data(), &TorrentCreatorDialog::piecesCountCalculated, Qt::QueuedConnection, count);
+        }
```

Comment:
I don't think you need to check it. And you can just call it normally:
```suggestion
        piecesCountCalculated(count);
```

Reply:
https://github.com/qbittorrent/qBittorrent/pull/23497#discussion_r2532119468 is unresolved.